hazelcast/src/main/java/com/hazelcast/internal/partition/MigrationEventType.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition;
+
+
+/**
+ * Migration event types.
+ * @see MigrationEventHandler
+ */
+public enum MigrationEventType {
```

Comment:
I am not really sure about this but, I think,  `MigrationEventType` class is not necessary. We are using enum classes like this when we want to expose them to the API. In this case, an `static int` on MessageTask/PartitionServiceProxy could do the job as well. 
Even if this stays, it should be moved to the `client` package as this is a specific type for `client` only. 
More specifically `com.hazelcast.client.impl` should be ok.

Reply:
Yes, this enum is no longer needed if we remove `MigrationEventHandler`